packages/devtools_app/lib/src/config_specific/flutter/import_export/import_export.dart

```diff
@@ -49,6 +56,52 @@ class ImportController {
         return;
     }
   }
+
+  void _navigateToImportScreen(DevToolsScreenType screenType) {
+    final availableScreens = scaffoldState.widget.tabs;
+    int screenIndex =
+        availableScreens.indexWhere((screen) => screen.type == screenType);
+    if (screenIndex == -1) {
+      screenIndex = scaffoldState.widget.tabs.length;
+      scaffoldState.widget.tabs.add(screenType.create());
+    }
+    scaffoldState.tabController.animateTo(screenIndex);
+  }
+
+  void _importTimeline(Map<String, dynamic> json) async {
+    OfflineData offlineData;
+    final timelineMode =
+        json[TimelineData.timelineModeKey] == TimelineMode.full.toString()
+            ? TimelineMode.full
+            : TimelineMode.frameBased;
+    if (timelineMode == TimelineMode.frameBased) {
+      offlineData = OfflineFrameBasedTimelineData.parse(json);
+    } else {
+      offlineData = OfflineFullTimelineData.parse(json);
+    }
+
+    if (offlineData.isEmpty) {
+      _notifications.push('Imported file does not contain timeline data.');
+      return;
+    }
+
+    // TODO(kenz): handle imports when we don't have any active controllers
+    // (i.e. when the connect screen is showing).
+    final timelineController = _controllers?.timeline;
```

Comment:
If this is the only controller the class needs, I recommend only providing this one.

Reply:
this is the only controller _for now_, but we will need other screen's controllers once those other screens support import. Given that, I think we can leave it as is.